### DISCUSSION TITLE
SIemens PLC

### DISCUSSION THREAD

**Discussion by Diemon1** [riclolsen/json-scada#19] [Q&A] [answered]
Hi Ricardo!
I'm a PLC engineer, and found your OSHMI project (and then json-scada). It seemed very interesting to me.

The only difference, that I'm working in food & bewerage industry, thus we dont have DNP3/IEC850 protocols here. Instead we have general PLC's (mainly S7-1500 from Siemens). But idea to have "hmi-in-smartphone" is cross-industry so i want to test this feature in some of my project.
As i can see Siemens S7 protocol implementation is not ready yet; but i see it in roadmap. Should i wait for it soon?
Thanks!

P.S. Greetings from Ukraine
P.P.S. привет joyshmitz! Как там Бандурка поживает? :) Вы на заводе подстанцию делаете?

**Comment by riclolsen** [comment, marked as answer]
Hi Diemon,

Thanks for the interest in the project, very nice to hear from someone from the food/beverage industry!

The project intends to fully support PLCs like S7 and others also, it begins by supporting energy protocols because it is my background and those are necessary for me to put it in production.

I do not have any dates yet but I intend to rely on the PLC4X project to support PLCs in general

* https://github.com/apache/plc4x

I do not intend to use any Java implementations, however. I can write drivers with Go or C# they also promise to support.

Could you monitor the PLC4x project and do some testing with your PLCs? This can be very helpful.

**Reply by joyshmitz** [reply]
Hi, @Diemon1 мы делаем.   Вам из турбинного зала Бандурка виднее :)

**Reply by Diemon1** [reply]
What's the reason not to use Java?

**Reply by riclolsen** [reply]
Just because I do not program in Java and also do not want to learn, but if somebody else wants to do this integration in Java that's ok.
